endpoints-framework/src/main/java/com/google/api/server/spi/response/ServletResponseResultWriter.java

```diff
@@ -124,8 +124,14 @@ protected void write(int status, Map<String, String> headers, String content) th
     // write response body
     if (content != null) {
       servletResponse.setContentType(SystemService.MIME_JSON);
-      servletResponse.setContentLength(content.getBytes("UTF-8").length);
-      servletResponse.getWriter().write(content);
+      if (streamResponse) {
+        objectWriter.writeValue(servletResponse.getWriter(), content);
+        servletResponse.getWriter().close();
```

Comment:
Shouldn't this be done even if there is an Exception on the previous line ?
This is what is done with writeValueAsString : (ClassUtil.closeOnFailAndThrowAsIOE(gen, e)).

Generally speaking, I think it would be nice to wrap everything the same way it is done by writeValueAsString to have similar exception handling + close/clean behaviour.

Reply:
I don't close the writer anymore (actually not useful).